Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst

```diff
@@ -0,0 +1,191 @@
+.. include:: /Includes.rst.txt
+.. index:: LinkBrowser; Custom
+.. _tutorial-github-link-handler:
+
+============================
+Create a custom link browser
+============================
+
+In this tutorial we create a custom link browser tab and the associated
+:ref:`backend link handler <tutorial_backend_link_handler>`.
+
+We create a new tab in the link browser window in the TYPO3 backend:
+
+..  include:: /Images/ManualScreenshots/Backend/CustomLinkBrowser.rst.txt
+
+..  tip::
+    If you want to link to a record in a custom table, configure the
+    :ref:`RecordLinkBrowser <TableRecordLinkBrowserTutorials>`. You do not
+    need a custom link browser in that scenario.
+
+We introduce a :ref:`custom link format <tutorial-core-link-handler>` to store
+the links in: `t3://github?issue=123`.
+
+This enables us to edit an existing link in the
+link browser or to change parts of the GitHub URI programmatically later on.
+
+And we :ref:`render the new link in the frontend <tutorial-typolink-builder>`
+automatically.
+
+..  contents::
+    :local:
+
+.. _tutorial_backend_link_handler:
+
+1. Create a link browser tab
+============================
+
+To create a link browser tab we implement the interface
+:php:`\TYPO3\CMS\Backend\LinkHandler\LinkHandlerInterface`.
+
+All backend link handlers provided by the Core extend the abstract class
+:php:`TYPO3\CMS\Backend\LinkHandler\AbstractLinkHandler`. However, this class is
+marked as :php:`@internal` and therefore can be changed by the Core at any time.
+
+You have the choice of implementing the :php:`LinkHandlerInterface` your self
+having a look at the :php:`AbstractLinkHandler` for best practices or to extend
+the :php:`AbstractLinkHandler`. In the latter case your code might break on
+updates though.
+
+In this tutorial we implement the :php:`LinkHandlerInterface` directly as it is
+best practice not to rely on internal classes.
+
+You can find the complete class in the extension EXT:examples on GitHub:
+`GitHubLinkHandler <https://github.com/TYPO3-Documentation/t3docs-examples/blob/main/Classes/LinkHandler/GitHubLinkHandler.php>`__.
+
+We will explain some of the important methods bellow:
+
+Initialization and dependencies
+-------------------------------
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerInitialize.rst.txt
+
+For technical reasons not all dependencies needed by the backend link handler can
+be acquired by :ref:`DependencyInjection`. Therefore the following two methods
+are called by Core classes once the dependencies are available:
+
+:php:`LinkHandlerInterface::initialize()` takes care of setting the
+:php:`\TYPO3\CMS\Backend\Controller\AbstractLinkBrowserController`, the identifier and
+the configuration information. In this example we only need the configuration,
+the other parameters might be needed in different scenarios. The
+:php:`\TYPO3\CMS\Core\Page\PageRenderer` has also to be created here for as it
```

Comment:
Why is the PageRenderer needed?

Reply:
to register the JavaScript, see render method. Shall we put this information here as well?